### PR TITLE
String literal types

### LINKED ISSUE #128: Add documentation for string literal types

### PR DESCRIPTION
**Pull request by sandersn**
A very short overview based on What's New in Typescript 1.8.

Fixes #128

### CODE REVIEW

**Comment by sandersn** [comment]
@DanielRosenwasser you should probably take a look to make sure that I hit the important points.

File: pages/Advanced Types.md [hunk 1]
```diff
@@ -315,6 +315,51 @@ Because [an ideal property of software is being open to extension](https://en.wi
...
+}
+
+let button = new UIElement();
+button.animate(0, 0, "ease-in");
+button.animate(0, 0, "uneasy");
+```
+
+You can pass any of the three allowed strings, but any other string will give the error
+
+```ts
```
Comment (DanielRosenwasser) [inline pages/Advanced Types.md]: This is text, not TypeScript

File: pages/Advanced Types.md [hunk 2]
```diff
@@ -315,6 +315,51 @@ Because [an ideal property of software is being open to extension](https://en.wi
...
+        }
+        else {
+            // error! should never reach here!
+        }
+    }
+}
+
+let button = new UIElement();
+button.animate(0, 0, "ease-in");
+button.animate(0, 0, "uneasy");
```
Comment (DanielRosenwasser) [inline pages/Advanced Types.md]: Nice. Can you add a comment like `// error:  '"uneasy"' is not assignable to '"ease-in" | "ease-out" | "ease-in-out"'`

File: pages/Advanced Types.md (hunk 1 above)
Comment (sandersn) [inline pages/Advanced Types.md]: done

File: pages/Advanced Types.md (hunk 2 above)
Comment (sandersn) [inline pages/Advanced Types.md]: done

**Comment by mhegazy** [comment]
looks good. @DanielRosenwasser any comments.

**Comment by mhegazy** [comment]
:+1:

File: pages/Advanced Types.md [hunk 3]
```diff
@@ -343,6 +343,51 @@ Because [an ideal property of software is being open to extension](https://en.wi
...
+    animate(dx: number, dy: number, easing: Easing) {
+        if (easing === "ease-in") {
+            // ...
+        }
+        else if (easing === "ease-out") {
+        }
+        else if (easing === "ease-in-out") {
+        }
+        else {
+            // error! should never reach here!
```
Comment (DanielRosenwasser) [inline pages/Advanced Types.md]: Well, `null` and `undefined`...

**Comment by DanielRosenwasser** [comment]
:+1:

**Pull request merged**